extensions/amp-lightbox/1.0/lightbox.js

```diff
@@ -0,0 +1,239 @@
+ * @param {{current: (!LightboxDef.LightboxApi|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function LightboxWithRef(
+  {
+    id,
+    animateIn = 'fade-in',
+    closeButtonAriaLabel,
+    children,
+    onOpen,
+    initialOpen,
+    opt_beforeOpen,
+    opt_afterClose,
+    ...rest
+  },
+  ref
+) {
+  // `show` is a trigger variable for animations.
+  // When the user opens/closes the lightbox, `show` is changed, which then triggers the appropriate animation and changes `visible` to true/false
+  // `visible` controls the actual visibility of the lightbox element on the page.
+  // When `visible` is true, the element exists on the page, else, it doesn't.
+  // Two states are necessary here in order to control the timing of the animation properly, else rendering conflicts with animation timing.
+  const [show, setShow] = useState(initialOpen);
+  const [visible, setVisible] = useState(false);
+  const styleClasses = useStyles();
+  const lightboxRef = useRef();
+  const beforeOpen = useCallback(
+    () => (opt_beforeOpen ? opt_beforeOpen() : undefined),
+    [opt_beforeOpen]
+  );
+  const afterClose = useCallback(
+    () => (opt_afterClose ? opt_afterClose() : undefined),
+    [opt_afterClose]
+  );
+
+  // We are using refs here to refer to common strings, objects, and functions used.
+  // This is because they are needed within `useEffect` calls below (but are not depended for triggering)
+  // This bypasses the exhaustive-deps eslint error
+  const onOpenFnRef = useRef();
+  onOpenFnRef.current = onOpen;
+  const animateInRef = useRef();
+  animateInRef.current = animateIn;
+  const lightboxContainWrapperRef = useRef();
+  lightboxContainWrapperRef.current = `.${styleClasses.lightboxContainWrapper}`;
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      open: () => {
+        beforeOpen();
+        setShow(true);
+      },
+      close: () => {
+        setShow(false);
+      },
+    }),
+    [beforeOpen]
+  );
+
+  useEffect(() => {
+    if (show) {
+      setVisible(true);
+    } else {
+      getRefElement(lightboxRef).then((element) => {
+        element = element.current;
+        if (element != undefined) {
+          const animation = element.animate(
+            ANIMATION_PRESETS[animateInRef.current],
+            {
+              duration: ANIMATION_DURATION,
+              direction: 'reverse',
+              fill: 'both',
+              easing: 'ease-in',
+            }
+          );
+          animation.onfinish = (opt_anim) => {
+            setVisible(false);
+            setStyle(
+              opt_anim.currentTarget.effect.target,
+              'opacity',
+              ANIMATION_PRESETS[animateInRef.current][0].opacity
+            );
+            setStyle(
+              opt_anim.currentTarget.effect.target,
+              'visibility',
+              ANIMATION_PRESETS[animateInRef.current][0].visibility
+            );
+            afterClose();
+          };
+          return () => {
```

Comment:
This doesn't do anything. The cleanup function has to be returned from the `useEffect`. Here you're returning it from some other callback inside the promise callback.